protos/flyteidl/core/catalog.proto

```diff
@@ -41,3 +41,17 @@ message CatalogMetadata {
        TaskExecutionIdentifier source_task_execution = 3;
     }
 };
+
+// Indicates the status of a catalog reservation operation.
+enum CatalogReservationStatus {
```

Comment:
You may want to move this into a message... because of the way code generation happens, this will end up with weird naming in some languages...

Reply:
@EngHabu  We'll want to move the CatalogCacheStatus enum at the top of this file then as well. Do we just want to move all of this file out of the core package and into the datacatalog package?

Reply:
Example: https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/core/execution.proto

I don't think you need to do the refactoring right now... we need a bigger project to revisit DataCatalog design and we can do the cleanup as part of that... but at least for new additions (like this PR), let's prevent future cleanup

Reply:
sounds great, done.